python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py

```diff
@@ -127,35 +129,19 @@ def get_compile_filters(parsed_args: Namespace) -> List[str]:
 
         return dbt_compile_options
 
-    def _get_dbt_nodes_and_dependencies(
-        self,
-    ) -> Tuple[Mapping[str, Any], Mapping[str, FrozenSet[str]]]:
-        """
-        For a given dbt Cloud job, fetch the latest run's dependency structure of executed nodes.
-        """
-        # Fetch information about the job.
-        job = self._dbt_cloud.get_job(job_id=self._job_id)
-        self._project_id = job["project_id"]
-        self._has_generate_docs = job["generate_docs"]
-
-        # We constraint the kinds of dbt Cloud jobs that we support running.
-        #
-        # A simple constraint is that we only support jobs that run multiple steps,
-        # but it must contain one of either `dbt run` or `dbt build`.
-        #
-        # As a reminder, `dbt deps` is automatically run before the job's configured commands.
-        # And if the settings are enabled, `dbt docs generate` and `dbt source freshness` can
-        # automatically run after the job's configured commands.
-        #
-        # These commands that execute before and after the job's configured commands do not count
-        # towards the single command constraint.
-        self._job_commands = job["execute_steps"]
-        self._job_materialization_command_step = (
-            DbtCloudCacheableAssetsDefinition.get_job_materialization_command_step(
-                execute_steps=self._job_commands
-            )
+    def _get_cached_compile_dbt_cloud_job_run(self, compile_run_id: int) -> Tuple[int, int]:
+        compile_run = self._dbt_cloud.get_run(
+            run_id=compile_run_id, include_related=["trigger", "run_steps"]
         )
+        compile_run_has_generate_docs = compile_run["trigger"]["generate_docs_override"]
+
+        compile_job_materialization_command_step = len(compile_run["run_steps"])
+        if compile_run_has_generate_docs:
```

Comment:
why this method of figuring out the compile step rather than parsing the commands? fine for now, but it does seem a bit roundabout.

Reply:
hahaha ok fair enough